graspy/align/orthogonal_procrustes.py

```diff
@@ -0,0 +1,117 @@
+
+import numpy as np
+from sklearn.utils import check_array
+from scipy.linalg import orthogonal_procrustes
+
+from .base import BaseAlign
+
+
+class OrthogonalProcrustes(BaseAlign):
+    """
+    Compute the matrix solution of the orthogonal Procrustes problem, which is
+    that given two matrices X and Y of equal shape, find an orthogonal matrix
+    that most closely maps X to Y.
+
+    Note that when used to match two datasets, this method unlike
+    SeedlessProcrustes, not only requires that the datasets have the same
+    number of entries, but also that there is some correspondence between the
+    vertices. In statistical spectral graphs, this usually corresponds to the
+    assumption that the vertex i in graph X has the same latent position as the
+    vertex i in graph Y.
+
+    Implementation-wise, this class is a wrapper of the
+    `scipy.linalg.orthogonal_procrustes`, which itself uses an algorithm
+    described in to find the optimal solution algorithm [1]_.
+
+    Parameters
+    ----------
+        freeze_Y : boolean, optional (default=False)
+            Currently irrelevant since the orthogonal transformation always
+            modifies only the first dataset.
+
+    Attributes
+    ----------
+        Q_X : array, size (d, d)
+              final orthogonal matrix, used to modify X
+
+        Q_Y : array, size (d, d)
+              final orthogonal matrix, used to modify Y
+              in OrthogonalProcrustes Q_Y is always equal to identity I
+
+    References
+    ----------
+    .. [1] Peter H. Schonemann, "A generalized solution of the orthogonal
+           Procrustes problem", Psychometrica -- Vol. 31, No. 1, March, 1996.
+
+    """
+
+    def __init__(
+        self,
+        freeze_Y=False,
+    ):
+
+        super().__init__(freeze_Y=freeze_Y)
+
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn matrices Q_X and Q_Y.
+        In regular orthogonal procrustes Q_X is a solution to the orthogonal
+        procrustes problem, and Q_Y is the identity matrix.
+
+        Parameters
+        ----------
+        X: np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y: np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        self: returns an instance of self
+        """
+
+        # check for numpy-ness, 2d-ness and finite-ness
+        if not isinstance(X, np.ndarray):
+            msg = f"first dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        if not isinstance(Y, np.ndarray):
+            msg = f"first dataset is a {type(Y)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X, accept_sparse=True, copy=True)
```

Comment:
why `accept_sparse = True` here? If it works for sparse mats, great, but has this been tested? Also, looks like the lines above wouldn't allow a sparse matrix input

Reply:
hm... yeah, you're right it serves no purpose. i think i wrote it before the check_array discussion we had in the previous pr. it honestly doesn't matter if we accept sparse, because for our purposes these are embeddings and they really shouldn't be sparse. i'll remove it

Reply:
resolved (via removing).